force-app/main/default/lwc/appModal/appModal.js

```diff
@@ -24,6 +24,9 @@ export default class AppModal extends LightningElement {
 	@api
 	modalSize;
 
+	@api
+	hideCloseButton = false;
```

Comment:
"hide" or "show"  prefixes are more like action names, may be acceptable

isCloseButtonHidden =  false;